src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -42,11 +41,31 @@ internal static class MetadataUtilities
             // Windows.*.winmd into a single fake Windows.winmd at runtime.
             // All other (application) winmds are left as is.
             var runtimeWinMdBuilder = ArrayBuilder<ModuleMetadata>.GetInstance();
+            AssemblyIdentity corLibrary = null;
             foreach (var block in metadataBlocks)
             {
                 var metadata = ModuleMetadata.CreateFromMetadata(block.Pointer, block.Size, includeEmbeddedInteropTypes: true);
                 try
                 {
+                    var reader = metadata.MetadataReader;
+                    if (corLibrary == null)
+                    {
+                        bool hasNoAssemblyRefs = reader.AssemblyReferences.Count == 0;
+                        // .NET Native uses a corlib with references.
+                        if (hasNoAssemblyRefs || metadata.Name.Equals("System.Private.CoreLib.dll", StringComparison.OrdinalIgnoreCase))
+                        {
+                            // If this assembly declares System.Object, assume it is the corlib.
+                            if (reader.DeclaresTheObjectClass())
+                            {
+                                corLibrary = reader.ReadAssemblyIdentityOrThrow();
```

Comment:
Yes, we'll choose the first assembly that meets the corlib requirements, even if there are multiple. That should be acceptable for evaluating expressions in the EE. Added a comment indicating that.